papermill/iorw.py

```diff
@@ -88,6 +88,11 @@ def listdir(cls, path):
 
     @classmethod
     def write(cls, buf, path):
+        dir = os.path.dirname(path)
+        if not dir.endswith("/"):
+            dir += "/"
+        if not os.path.exists(dir):
+            raise Exception('output folder {} doesn\'t exist!'.format(dir))
```

Comment:
I'd rewrite this block as:
```python
dir_name = os.path.dirname(path)
if not os.path.exists(dir_name):
    raise NoSuchDirectory("The output folder {} does not exist.".format(dir_name))
```

and define a `NoSuchDirectory` exception

Reply:
@betatim have you added this NoSuchDirectory exception or shall I add and commit the code?

Reply:
@betatim added the code block and committed.